app/models/school.rb

```diff
@@ -60,6 +67,17 @@ class School < ApplicationRecord
     end
   end
 
+  def self.search_by_name(name)
+    scope = search_by_fields(name).limit(NAME_SEARCH_LIMIT)
+    NAME_SYNONYMS.find do |key, value|
+      if name&.downcase&.include?(key)
+        synonym_name = name.downcase.gsub(key, value)
+        return scope + search_by_fields(synonym_name).limit(NAME_SEARCH_LIMIT)
+      end
+    end
+    scope
```

Comment:
If I've got this right 

`.search_by_name("Some First School")` 

will look for 

`search_by_fields("Some First School") + search_by_fields("Some Firsaint School")`

I don't think the extra search would return any results but it doesn't seem quite right

I think this needs to be a regexp

```ruby
if name&.downcase&.match?(%r{\b#{key}\b}i)
```